src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -1,16 +1,41 @@
 import { useState, useEffect } from 'react'
 import { useGetTriggersQuery } from '../../utils/generated/graphql'
 
-type triggerTypes = 'applicationTrigger' | 'reviewTrigger'
+type TriggerType = 'applicationTrigger' | 'reviewAssignmentTrigger' | 'reviewTrigger'
 
-const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTypes }) => {
-  const { serialNumber, trigger } = props
+interface TrigerQueryProps {
+  serialNumber?: string
+  reviewAssignmentId?: number
+  reviewId?: number
+  triggerType?: TriggerType
+}
+
+const useTriggerProcessing = (props: TrigerQueryProps) => {
+  const { serialNumber, reviewAssignmentId, reviewId } = props
   const [isProcessing, setIsProcessing] = useState(true)
   const [triggerError, setTriggerError] = useState(false)
 
+  // Ensure at least one the primary identifiers is provided
+  if (!serialNumber && !reviewAssignmentId && !reviewId) {
+    console.log('INVALID QUERY')
+    setIsProcessing(false)
+    setTriggerError(true)
+  }
+
+  // If triggerType not provided, infer it from the supplied ID
+  const triggerType = props.triggerType
+    ? props.triggerType
+    : serialNumber
+    ? 'applicationTrigger'
+    : reviewAssignmentId
+    ? 'reviewAssignmentTrigger'
+    : 'reviewTrigger'
```

Comment:
I think this and the `type` defined up here would be better separated in its own helper method, just to be clear what's going on and helping to read this code. The method could be in the same file, if that is simpler.